File: app/src/main/java/com/example/bottlr/LocationAdapter.java
package com.example.bottlr;

import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.TextView;
import androidx.annotation.NonNull;
import androidx.recyclerview.widget.RecyclerView;
import java.util.ArrayList;
import java.util.List;

public class LocationAdapter extends RecyclerView.Adapter<LocationAdapter.LocationViewHolder> {
    private final List<Location> locations;

    public LocationAdapter(List<Location> locations) {
        this.locations = locations;
    }

    @NonNull
    @Override
    public LocationViewHolder onCreateViewHolder(@NonNull ViewGroup parent, int viewType) {
        View view = LayoutInflater.from(parent.getContext()).inflate(R.layout.location_item, parent, false);
        return new LocationViewHolder(view);
    }

    @Override
    public void onBindViewHolder(@NonNull LocationViewHolder holder, int position) {
        Location location = locations.get(position);
        holder.bind(location);
    }

    @Override
    public int getItemCount() {
        return locations.size();
    }

    static class LocationViewHolder extends RecyclerView.ViewHolder {
        private final TextView locationName;
        private final TextView locationCoordinates;
        private final TextView locationTimestamp;

        public LocationViewHolder(@NonNull View itemView) {
            super(itemView);
            locationName = itemView.findViewById(R.id.location_name);
            locationCoordinates = itemView.findViewById(R.id.location_coordinates);
            locationTimestamp = itemView.findViewById(R.id.location_timestamp);
        }

        public void bind(Location location) {
            locationName.setText(location.getName());
            locationCoordinates.setText(location.getGpsCoordinates());
            locationTimestamp.setText(location.getTimeDateAdded());
        }
    }
}

File: app/src/main/kotlin/com/bottlr/app/LocationAdapter.kt
package com.bottlr.app

import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.widget.Button
import android.widget.TextView
import androidx.recyclerview.widget.RecyclerView

class LocationAdapter(
    private var locations: List<Location>,
    private var allLocations: List<Location>,
    private val onLocationClick: OnLocationCheckListener
) : RecyclerView.Adapter<LocationAdapter.LocationViewHolder>() {

    fun interface OnLocationCheckListener {
        fun onButtonClick(name: String?, coordinates: String?, date: String?)
    }

    class LocationViewHolder(itemView: View) : RecyclerView.ViewHolder(itemView) {
        val textViewName: TextView = itemView.findViewById(R.id.textViewName)
        val textViewCoordinates: TextView = itemView.findViewById(R.id.textViewCoordinates)
        val textViewTimestamp: TextView = itemView.findViewById(R.id.textViewTimestamp)
        val locationButton: Button = itemView.findViewById(R.id.locationsinglebutton)
    }

    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): LocationViewHolder {
        val view = LayoutInflater.from(parent.context).inflate(R.layout.location_item, parent, false)
        return LocationViewHolder(view)
    }

    override fun onBindViewHolder(holder: LocationViewHolder, position: Int) {
        val location = locations[position]
        holder.textViewName.text = location.name
        holder.textViewCoordinates.text = location.gpsCoordinates
        holder.textViewTimestamp.text = location.timeDateAdded
        holder.locationButton.setOnClickListener {
            onLocationClick.onButtonClick(location.name, location.gpsCoordinates, location.timeDateAdded)
        }
    }

    override fun getItemCount(): Int = locations.size
}
